var/spack/repos/builtin/packages/nsimd/package.py

```diff
@@ -85,6 +85,38 @@ def generate_code(self, spec, prefix):
     def cmake_args(self):
         # Required SIMD argument
         simd = self.spec.variants['simd'].value
+        if simd == 'auto':
+            # x86
+            if 'avx512' in self.spec.target:
+                simd = 'AVX512_SKYLAKE'
+            elif self.spec.satisfies('target=mic_knl'):
+                simd = 'AVX512_KNL'
+            elif 'avx2' in self.spec.target:
+                simd = 'AVX2'
+            elif 'avx' in self.spec.target:
+                simd = 'AVX'
+            elif 'sse4_2' in self.spec.target:
+                simd = 'SSE42'
+            elif 'sse2' in self.spec.target:
+                simd = 'SSE2'
+            # ARM
+            elif 'sve' in self.spec.target:
+                # We require an explicit choice for particluar bit widths
+                simd = 'SVE'
+            elif self.spec.satisfies('target=aarch64'):
+                simd = 'AARCH64'
+            elif 'neon' in self.spec.target:
+                simd = 'NEON128'
+            # POWER
+            elif 'vsx' in self.spec.target:
+                simd = 'VSX'
+            elif (self.spec.satisfies('target=ppc64') or
+                  self.spec.satisfies('target=ppc64le')):
```

Comment:
I think there is a typo here:
```suggestion
            elif (self.spec.satisfies('target=ppc64:') or
                  self.spec.satisfies('target=ppc64le:')):
```
What I proposed is "generic ppc64 or higher", so it includes power8, power9 etc. What is currently encoded is "exactly matches generic ppc64".

Reply:
Oh; since you said "there might be typos" I thought that was one. I'll update.

Reply:
Done.